library/src/hcc_detail/hipsolver.cpp

```diff
@@ -2902,6 +2958,664 @@ catch(...)
     return exception2hip_status();
 }
 
+/******************** SYEVD/HEEVD ********************/
+hipsolverStatus_t hipsolverSsyevd_bufferSize(hipsolverHandle_t   handle,
+                                             hipsolverEigMode_t  jobz,
+                                             hipsolverFillMode_t uplo,
+                                             int                 n,
+                                             float*              A,
+                                             int                 lda,
+                                             float*              W,
+                                             int*                lwork)
+try
+{
+    size_t sz;
+
+    rocblas_start_device_memory_size_query((rocblas_handle)handle);
+    rocblas_status status = rocsolver_ssyevd((rocblas_handle)handle,
+                                             hip2rocblas_evect(jobz),
+                                             hip2rocblas_fill(uplo),
+                                             n,
+                                             nullptr,
+                                             lda,
+                                             nullptr,
+                                             nullptr,
+                                             nullptr);
+    rocblas_stop_device_memory_size_query((rocblas_handle)handle, &sz);
+
+    // space for E array
+    sz += sizeof(float) * n;
```

Comment:
And what would happen if a user never calls this function because it wants to use rocsolver automatic reallocation?
How would you allocate space for E?